MM-9990: Ephemeral Post Documentation

DSchalla opened the pull request:
This implements the documentation for change MM-9990 from https://github.com/mattermost/mattermost-server/pull/8611. Since the request body expects a structure that didn't exist as a schema I was a bit uncertain how to implement it, so if you'd prefer to create a dedicated schema for it or another proposal I am more than happy to change it.

jwilander commented on v4/source/posts.yaml:
```diff
@@ -47,6 +47,47 @@
+      description: |
+        Create a new ephemeral post in a channel.
+        ##### Permissions
+        Must have `create_post_ephemeral` permission (currently only given to system adin)
```
Typo `adin` -> `admin`

jwilander commented on v4/source/posts.yaml:
```diff
@@ -47,6 +47,47 @@
+                required:
+                  - channel_id
+                  - message
+                description: Post object to create
```
This might work, but I'll think you'll have to specify properties for `channel_id` and `message`

DSchalla replied:
Worked fine, but you are right, it should be added never the less!

DSchalla commented:
Thanks, fixed the typo and implemented the parameter descriptions. Build looks fine:
```
Building mattermost openapi yaml for v4
v4/html/static/mattermost-openapi-v4.yaml is valid
Complete
```
